src/sagemaker/workflow/step_collections.py

```diff
@@ -54,6 +54,13 @@ def properties(self):
         size = len(self.steps)
         return self.steps[size - 1].properties
 
+    def _find_step_dependencies(self):
+        """Find the step names this step collection is dependent on."""
+        dependencies = {}
+        for step in self.steps:
+            dependencies[step.name] = step._find_step_dependencies()
+        return dependencies
+
```

Comment:
This [merged commit](https://github.com/aws/sagemaker-python-sdk/commit/79ecab28a5004057c0842cfa291ef964d1f3129b#diff-043e8c79b0b9ccd41f8164a29ccc1f9e67f38e349aafffb96cd91762c3824bc7R49-R56) just supports StepCollection properties.
What if another step is referencing the StepCollection properties? Would the last sub step name be appended in the adjacency list?
Please point me to a test of this case if any